modules/searcher/MixCloudSearcher.js

```diff
@@ -0,0 +1,32 @@
+const request = require('request');
+const MixCloudTrack = require('../../models/track/MixCloudTrack');
+
+function MixCloudSearcher() {
+  //
+}
+
+MixCloudSearcher.prototype.search = function(keyword, limit) {
+  let limit = limit || 20;
+  let queryURL = `https://api.mixcloud.com/search/?limit=${limit}&q=${keyword}&type=cloudcast`;
+
+  return new Promise((resolve, reject) => {
+    request(queryURL, (err, response, rawTracks) => {
+      if (err) {
+        console.error(err);
+        reject(err);
+      } else {
+        const parseRawTracks = JSON.parse(rawTracks);
+        const mixCloudTracks = parseRawTracks['data'].map(rawTrack => {
```

Comment:
after `JSON.parse`, they are not string anymore and will be transformed into real Object, so the naming here maybe confusing